tests/jsonb_field_testing/tests.py

```diff
@@ -391,3 +391,78 @@ def test_large_search(self):
+        # Method should not raise an exception
+        val = FilterTree.text_similarity_filter(['root', 'key'], '✓', path_multiple=True)
+        self.assertIsNotNone(val)
+
+    def test_unicode_search_value(self):
+        """Ensure that unicode values can be filtered for successfully."""
+        record = JsonBModel.objects.create(data={"key": "'Verified ✓'"})
+
+        filt = {'key': {'_rule_type': 'containment', 'pattern': '✓'}}
+        query = JsonBModel.objects.filter(data__jsonb=filt)
+        self.assertEqual(list(query), [record])
+
+    def test_unicode_search_value_multiple(self):
+        """Ensure that unicode values can be filtered for successfully."""
+        record = JsonBModel.objects.create(data={"root": {"key": "'Verified ✓'"}})
+
+        filt = {'root': {'key': {'_rule_type': 'containment_multiple', 'pattern': '✓'}}}
+        query = JsonBModel.objects.filter(data__jsonb=filt)
+        self.assertEqual(list(query), [record])
+
+    def test_unicode_search_key(self):
+        """Ensure that unicode keys can be filtered on successfully."""
+        record = JsonBModel.objects.create(data={"root": {"✓": "'Verified'"}})
+
+        filt = {'root': {'✓': {'_rule_type': 'containment', 'pattern': 'Verified'}}}
+        query = JsonBModel.objects.filter(data__jsonb=filt)
+        self.assertEqual(list(query), [record])
+
+    def test_unicode_search_key_multiple(self):
+        """Ensure that unicode keys can be filtered on successfully."""
+        record = JsonBModel.objects.create(data={"root": {"✓": "'Verified'"}})
+
+        filt = {'root': {'✓': {'_rule_type': 'containment_multiple', 'pattern': 'Verified'}}}
+        query = JsonBModel.objects.filter(data__jsonb=filt)
+        self.assertEqual(list(query), [record])
+
+    def test_split_search_pattern_single_words(self):
+        result = FilterTree.split_search_pattern('hello world')
+        self.assertEqual(result, ['hello', 'world'])
+
+    def test_split_search_pattern_double_quoted_words(self):
+        result = FilterTree.split_search_pattern('"hello world"')
+        self.assertEqual(result, ['hello world'])
+
+    def test_split_search_pattern_single_quoted_words(self):
+        result = FilterTree.split_search_pattern("'hello world'")
+        self.assertEqual(result, ['hello world'])
+
+    def test_split_search_pattern_mixed_words(self):
+        result = FilterTree.split_search_pattern('the "hello world" message')
+        self.assertEqual(result, ['the', 'hello world', 'message'])
+
+    def test_split_search_pattern_unpaired_quote_front(self):
+        result = FilterTree.split_search_pattern('"hello world')
+        # The unpaired quote is trimmed off
+        self.assertEqual(result, ['hello', 'world'])
+
+    def test_split_search_pattern_unpaired_quote_trail(self):
+        result = FilterTree.split_search_pattern('hello world"')
+        # The unpaired quote is trimmed off
+        self.assertEqual(result, ['hello', 'world'])
```

Comment:
Checked against `develop` to ensure we maintained historical behavior, all of these `test_split_search_pattern...` tests pass in `develop` except these last two for unpaired quotes. It appears in `develop` searching with an unpaired quotation mark will cause an unhandled exception:
```
======================================================================
ERROR: test_split_search_pattern_unpaired_quote_trail (tests.jsonb_field_testing.tests.JsonBFilterTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/grout/tests/jsonb_field_testing/tests.py", line 466, in test_split_search_pattern_unpaired_quote_trail
    result = FilterTree.split_search_pattern('hello world"')
  File "/opt/grout/grout/lookups.py", line 77, in split_search_pattern
    return shlex.split(pattern)
  File "/usr/local/lib/python2.7/shlex.py", line 279, in split
    return list(lex)
  File "/usr/local/lib/python2.7/shlex.py", line 269, in next
    token = self.get_token()
  File "/usr/local/lib/python2.7/shlex.py", line 96, in get_token
    raw = self.read_token()
  File "/usr/local/lib/python2.7/shlex.py", line 172, in read_token
    raise ValueError, "No closing quotation"
ValueError: No closing quotation
```
I was able to reproduce this effect in the DRIVER staging environment, which uses Grout.